core/src/main/java/org/mskcc/cbio/portal/servlet/ClinicalJSON.java

```diff
@@ -125,22 +133,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
         JSONArray maps = null;
 
         try {
-            if (samples == null || samples.equals(ALL) ) {
-                clinicals = DaoClinicalData.getData(cancerStudyId);
-                maps = clinicals2JSONArray(clinicals);
-            } else {
-                clinicals = DaoClinicalData.getData(cancerStudyId,
-                        Arrays.asList(samples.trim().split(SAMPLES_DELIMITER)));
-                maps = clinicals2JSONArray(clinicals);
-            }
+        	if(cancerStudyId != null) {
+        		CancerStudy cancerStudy = DaoCancerStudy
+                        .getCancerStudyByStableId(cancerStudyId);
+        		if (cancerStudy != null && accessControl.isAccessibleCancerStudy(cancerStudy.getCancerStudyStableId()).size() == 1) {
+        			if (samples == null || samples.equals(ALL) ) {
+                        clinicals = DaoClinicalData.getData(cancerStudyId);
+                        maps = clinicals2JSONArray(clinicals);
+                    } else {
+                        clinicals = DaoClinicalData.getData(cancerStudyId,
+                                Arrays.asList(samples.trim().split(SAMPLES_DELIMITER)));
+                        maps = clinicals2JSONArray(clinicals);
+                        
```

Comment:
What I mean is that the following code should be run in both cases:

for (ClinicalData c : clinicals) {
maps.add(reflectToMap(c));
 }